### PR TITLE
Delete columns

### LINKED ISSUE #45: Removing last column of the DataGrid flattens all of the data
**Actual behavior**
Removing the last column of the `DataGrid` flattens all of the data

**Expected behavior**
The data should remain intact and in the same layout

**To Reproduce**
1. Open the Tabular Data Editor extension from a CSV file
2. Right-click on the last column
3. Click `Remove Columns`

**Screenshots**

**Additional context**
N/A

### LINKED ISSUE #47: Deleting the last row of the editor produces strange behavior.
**Actual behavior**
When the last row of the editor gets deleted, the entire table gets pasted directly below the trimmed table.

**Expected behavior**
The final row is simply removed.

**To Reproduce**
1. Go to Jupyterlab
2. Open a test.csv with TabularDataEditor and attempt to remove the last row.

**Screenshots**
Screen Shot 2020-07-08 at 9 38 53 AM
Screen Shot 2020-07-08 at 9 39 00 AM
**Desktop (please complete the following information):**
 - iOS

**Additional context**

### PR DESCRIPTION
**Pull request by ghost**
# Description of the Pull Request

Modified some functions in `model.ts` and `widget.ts` to fix bugs occurring the datagrid.

### Issue being fixed:
Fixes #45 
Fixes #47. It also fixes a problem with the cells de-focusing after an edit.
### Changes proposed:

- In `_updateGrid` the `DataModel` doesn't get replaced if it already exists.
- In `_removeColumns` different behavior happens when the column being removed is the last one.
- In `_setField` the row index is shifted by 1. It was previously mistaking the second-to-last row as the last row.

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -57,6 +57,7 @@ export default class EditableDSVModel extends MutableDataModel {
       default:
         throw 'unreachable';
     }
+    console.log(model.rawData);
```
Comment (kgoo124) [inline src/model.ts:60]: Let's get rid of this

File: src/model.ts [hunk 2]
```diff
@@ -189,9 +190,13 @@ export default class EditableDSVModel extends MutableDataModel {
...
-    model.rawData =
-      model.rawData.slice(0, rowRemovedIndex) +
-      model.rawData.slice(rowAfterIndex);
+    if (!rowAfterIndex) {
+      model.rawData = model.rawData.slice(0, rowRemovedIndex);
+    } else {
+      model.rawData =
+        model.rawData.slice(0, rowRemovedIndex) +
+        model.rawData.slice(rowAfterIndex);
+    }
```
Comment (kgoo124) [inline src/model.ts:199]: What do you think about using a ternary? 
```
model.rawData = rowAfterIndex ?
 model.rawData.slice(0, rowRemovedIndex) + model.rawData.slice(rowAfterIndex) :
 model.rawData.slice(0, rowRemovedIndex)```

If this looks too confusing, then feel free to disregard this

File: src/model.ts [hunk 3]
```diff
@@ -204,19 +209,49 @@ export default class EditableDSVModel extends MutableDataModel {
...
+        endIndex =
+          model.getOffsetIndex(nextRow, nextCol) - shift - trailingRowDelimeter;
+      }
       diff = endIndex - startIndex;
       model.rawData =
         model.rawData.slice(0, startIndex) + model.rawData.slice(endIndex);
       shift += diff;
     }
+
     console.log(model.rawData);
```
Comment (kgoo124) [inline src/model.ts:255]: Let's remove this as well

File: src/model.ts [hunk 4]
```diff
@@ -94,7 +95,7 @@ export default class EditableDSVModel extends MutableDataModel {
     // if we are getting the last column.
     if (column === model.columnCount('body') - 1) {
       // Check if we are getting any row but the last.
-      if (row < model.rowCount('body') - 1) {
+      if (row <= model.rowCount('body') - 1) {
```
Comment (kgoo124) [inline src/model.ts:98]: I think we should just go with `row < model.rowCount('body')` instead of adding the equals sign. Just for clarity

**Review by thegooman — CHANGES_REQUESTED** [review]
Everything works well! Just had a few minor suggestions to help simplify and clarify the code! Good work!

File: src/model.ts (hunk 4 above)
Comment (lmcnichols) [reply src/model.ts:98]: Agreed.

File: src/model.ts (hunk 3 above)
Comment (lmcnichols) [reply src/model.ts:255]: It's gone.

File: src/model.ts (hunk 2 above)
Comment (lmcnichols) [reply src/model.ts:199]: Ternary looks fine to me. Cleaner. Shorter.

File: src/model.ts (hunk 1 above)
Comment (lmcnichols) [reply src/model.ts:60]: It's gone.

**Review by thegooman — APPROVED** [review]

**Pull request merged**